### ISSUE TITLE
[v2] can't exit on error with ^C if exitOnCtrlC: false

### ISSUE THREAD

**Issue by zkat** [vadimdemedes/ink#127]
Hi again! I'm trying out the new patches in `2.0.0-15` and found out that my custom `^C` handler, which just does `app.unmount()`, isn't able to unmount the app if an error happens -- in fact, there's no way to exit at all!

1. `git clone https://github.com/npm/libpickupdates`
2. `cd libpickupdates && git checkout zkat/exit-on-error`
3. `npm run build`
4. `node index.js`
5. Try to get out without forcibly closing your terminal.

There's also a probably-the-same bug in `latest`, where `^C` doesn't seem to work at all. Any clues?

**Comment by vadimdemedes** [comment]
Hm interesting, I'm at work now, but will investigate in about 5 hours.

**Comment by vadimdemedes** [comment]
Reproduced with:

```
const readline = require('readline')

const onKeypress = data => {
  console.log([data])
  
  if (data === '\u0003') {
    console.log('exit')
    process.stdin.removeListener('keypress', onKeypress)
    process.stdin.setRawMode(false)
  }
}

readline.emitKeypressEvents(process.stdin)
process.stdin.setRawMode(true)
process.stdin.on('keypress', onKeypress)
```

**Comment by vadimdemedes** [comment]
Figured it out, it needs `pause()` and `resume()` to let the process quit:

```js
const readline = require('readline')

const onKeypress = data => {
  console.log([data])
  
  if (data === '\u0003') {
    console.log('exit')
    process.stdin.removeListener('keypress', onKeypress)
    process.stdin.setRawMode(false)
    process.stdin.pause()
  }
}

readline.emitKeypressEvents(process.stdin)
process.stdin.setRawMode(true)
process.stdin.on('keypress', onKeypress)
process.stdin.resume()
```

**Comment by vadimdemedes** [comment]
It seems the same problem occurs when using `on('data')` too, so I will update Ink to `pause()` and `resume()` too, which should fix npm's CLI too!

**Comment by vadimdemedes** [comment]
I've also added tests to ensure processes using Ink always exit cleanly, without `process.exit()`, see https://github.com/vadimdemedes/ink/commit/3d477dbc0e35f47bb497ef72c1650b3cfaf9f625.

Released `2.0.0-16`, which should fix this issue!
